cmd/cloud-controller-manager/nodeipamcontroller.go

```diff
@@ -75,6 +75,12 @@ func startNodeIpamController(initContext app.ControllerInitContext, ccmConfig *c
 		return nil, false, nil
 	}
 
+	// Cannot run cloud ipam controller if cloud provider is nil (not set or 'external')
+	if cloud == nil && ccmConfig.ComponentConfig.KubeCloudShared.CIDRAllocatorType == string(ipam.CloudAllocatorType) {
+		klog.Warningf("--cidr-allocator-type is set to 'CloudAllocator' but cloud provider is not configured")
+		return nil, false, nil
```

Comment:
We can also error here and have kube-controller-manager exit, instead of just skipping the controller.

Reply:
I'm not familiar with this area, why `cloud-controller-manager` should not have a cloud provider, is not that implicit?

Reply:
Yeah technically speaking we don't _really_ need this check in cloud-controller-manager, it should always be set -- but I added it here for consistency with kube-controller-manager